### PR TITLE
kubelet: remove the deprecated --master-service-namespace flag

### PR DESCRIPTION
**Pull request by SataQiu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
The `--master-service-namespace` was deprecated since #44250 (v1.7), and the similar flag of kube-apiserver has been removed by #114446, it's time to remove the deprecated --master-service-namespace flag for kubelet.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
kubelet: the deprecated `--master-service-namespace` flag is removed in v1.27
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by SataQiu** [comment]
/assign @liggitt 
Please take a look if you have time

**Comment by liggitt** [comment]
/unassign
please route to node/kubelet reviewers/approvers to make sure we load-balance reviews

File: pkg/kubelet/kubelet_pods.go [hunk 1]
```diff
@@ -559,7 +559,7 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string, enableServiceLinks bool) (map[
 		// from the master service namespace, even if enableServiceLinks is false.
 		// We also add environment variables for other services in the same
 		// namespace, if enableServiceLinks is true.
-		if service.Namespace == kl.masterServiceNamespace && masterServices.Has(serviceName) {
+		if service.Namespace == metav1.NamespaceDefault && masterServices.Has(serviceName) {
```
Comment (marquiz) [inline pkg/kubelet/kubelet_pods.go:562]: nit: you could adjust the comment above accordingly(?)

Comment (SataQiu) [reply pkg/kubelet/kubelet_pods.go:562]: Thanks @marquiz Good catch! Updated.

**Comment by marquiz** [comment]
Thanks @SataQiu for the update. Codewise looks good to me
/lgtm

**Pull request merged**